src/helm/benchmark/window_services/huggingface_window_service.py

```diff
@@ -0,0 +1,39 @@
+from helm.proxy.clients.huggingface_tokenizer import HuggingFaceTokenizers
+from .local_window_service import LocalWindowService
+from .tokenizer_service import TokenizerService
+from helm.proxy.clients.huggingface_client import HuggingFaceModelConfig
+
+
+class HuggingFaceWindowService(LocalWindowService):
+    def __init__(self, service: TokenizerService, model_config: HuggingFaceModelConfig):
+        super().__init__(service)
+        self._tokenizer_name = model_config.model_id
+        tokenizer = HuggingFaceTokenizers.get_tokenizer(model_config.model_id)
+        self._prefix_token = tokenizer.bos_token
+        self._end_of_text_token = tokenizer.eos_token
+        self._max_request_length = tokenizer.model_max_length
+
+    @property
+    def max_sequence_length(self) -> int:
+        """Return the max sequence length of this tokenizer."""
+        return self._max_request_length
+
+    @property
+    def max_request_length(self) -> int:
+        """Return the max request length of this tokenizer."""
+        return self.max_sequence_length
+
+    @property
+    def end_of_text_token(self) -> str:
+        """The end of text token."""
+        return self._end_of_text_token
```

Comment:
nit: why not just forward the request to tokenizer.eos_token? (similarly for the rest)

Reply:
This makes explicit that we only use the tokenizer for fetching the metadata, and not for tokenization (which is done separately elsewhere).

Reply:
i.e. we don't keep a handle to the tokenizer in the instance outside of `__init__`